prov/gni/src/gnix_mr_notifier.c

```diff
@@ -55,69 +57,69 @@ notifier_verify_stuff(struct gnix_mr_notifier *mrn) {
 }
 
 int
-_gnix_notifier_init(struct gnix_mr_notifier *mrn)
+_gnix_notifier_init(void)
 {
-	if (mrn == NULL) {
-		GNIX_INFO(FI_LOG_MR, "mr notifier NULL\n");
-		return -FI_EINVAL;
-	}
-
-	mrn->fd = 0;
-	mrn->cntr = NULL;
-	fastlock_init(&mrn->lock);
+	global_mr_not.fd = -1;
+	global_mr_not.cntr = NULL;
+	fastlock_init(&global_mr_not.lock);
 
 	return FI_SUCCESS;
 }
 
 int
-_gnix_notifier_open(struct gnix_mr_notifier *mrn)
+_gnix_notifier_open(struct gnix_mr_notifier **mrn)
 {
 	int ret = FI_SUCCESS;
 	int kdreg_fd, ret_errno;
         kdreg_get_user_delta_args_t get_user_delta_args;
 
-	if ((mrn->fd != 0) || (mrn->cntr != NULL)) {
-		GNIX_INFO(FI_LOG_MR, "mr notifier already open\n");
-		return -FI_EBUSY;
-	}
+	fastlock_acquire(&global_mr_not.lock);
 
-	fastlock_acquire(&mrn->lock);
+	if (global_mr_not.fd < 0) {
```

Comment:
A single 'if' containing the already initialized code would remove all of the whitespace changes.

```C 
if (global_mr_not.fd >= 0)
   // skip all of this and return.
```

Additionally, it would simply the exit case if you did the following:

```C

if (not-initialized)
    goto done;
...
     //code for normal open
done:
    increment reference++
```
If the reference count was initialized to 0 in the init function, then ref_cnt ++ would have the same effect as 'ref_cnt = 1' here.

Much of this code is unchanged otherwise.